lib/firefox/webdriver/firefoxDelegate.js

```diff
@@ -45,6 +58,19 @@ class FirefoxDelegate {
       // TODO clear the original log file!
     }
 
+    if (this.firefoxConfig.geckoProfiler) {
+      let profileFilename = path.join(this.baseDir, 'geckoProfile.json');
+
+      // Must use String.raw below or else the backslashes on Windows will be escapes.
+      const script = `
+      var callback = arguments[arguments.length - 1];
+       Services.profiler.dumpProfileToFileAsync(String.raw\`${profileFilename}\`)
+        .then(callback)
+        .catch((e) => callback({'error' : e}));
+       `;
+      await runner.runPrivilegedAsyncScript(script, 'Stop GeckoProfiler');
```

Comment:
There's an API to add raw files to browsertime reports: look for `result.extraJson`.  Use that.  (But I would still have Gecko write the file; streaming 100s of Mb over Web Driver sounds like a bad idea.)

Reply:
I'm not entirely sure how to use this API.  It accepts JSON as input, and then it write that out into a separate file under url/pages/data and gzip's the file.  It seems useful if we had a small amount of JSON data to output but, like you said, I think it's better that Gecko writes the file out directly.  

Thinking longer term, we could use this to output some information after processing the profile in some way.  Maybe if you're interested in a list of top-10 hottest JS functions or something.

Reply:
Ah, I see, that API is not ideal.  But there's a `files` block in the results, like:
```
      if (this.options.chrome && this.options.chrome.collectPerfLog) {
        results.files.perfLog.push(
          `${pathToFolder(url, this.options)}chromePerflog-${index}.json.gz`
        );
      }
```
in the collector.  Did you already do this?  If not, next time you're in the area, try it out?  And gzip would be nice too.